CMake build fails with `error: inline assembly requires more registers than available`

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   PLEASE ENSURE ALL SECTIONS OF THIS TEMPLATE ARE FILLED IN AND THAT THERE ARE
   NO OTHER CHANGES TO THE TEMPLATE.    
   
   Only bugs should be raised here as issues. Questions or enhancements should instead be raised on 
   our forums:
   https://forums.mbed.com/ .

   *************************************************************************************

-->

### Description of defect

<!--
    Add detailed description of what you are reporting.
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html
-->

The build command
```
mbed-tools compile -t ARM -m DISCO_L072CZ_LRWAN1
```
fails with
```
../../../../mbed-os/connectivity/mbedtls/source/bignum.c:1563:9: error: inline assembly requires more registers than available                                                                                    
        MULADDC_INIT                                                                                                                                                                                              
        ^
```

Note: This failure happens when compiling Mbed TLS for any STM32L0 targets, with the Arm Compiler and mbed-tools.

#### Target(s) affected by this defect ?

DISCO_L072CZ_LRWAN1

#### Toolchain(s) (name and version) displaying this defect ?

Arm Compiler 6.14

#### What version of Mbed-os are you using (tag or sha) ?
<!--
    For a released version please provide the release tag (this can be found as per the instructions below)

    mbed-os version can be found in:
    https://github.com/ARMmbed/mbed-os/blob/master/platform/include/platform/mbed_version.h
    The tag can be reconstructed as follows:
    mbed-os-MBED_MAJOR_VERSION.MBED_MINOR_VERSION.MBED_PATCH_VERSION
     
-->

mbed-os-6.9.0

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)

mbed-tools: 7.8.0

#### How is this defect reproduced ?

Run the command in the description above.

--- a/connectivity/mbedtls/CMakeLists.txt
+++ b/connectivity/mbedtls/CMakeLists.txt
@@ -102,3 +102,20 @@ target_compile_definitions(mbed-mbedtls
     INTERFACE
         MBED_CONF_MBEDTLS_PRESENT=1
 )
+
+# Workaround for https://github.com/ARMmbed/mbedtls/issues/1077
+# which affects cores without __thumb2__ set by the compiler
+# due to the lack of full Thumb-2 support
+set(partial-thumb2-cores
+    Cortex-M0
+    Cortex-M0+
+    Cortex-M1
+    Cortex-M23
+    Cortex-M23-NS
+)
+if(MBED_CPU_CORE IN_LIST partial-thumb2-cores)
+    target_compile_definitions(mbed-mbedtls
+        INTERFACE
+            MULADDC_CANNOT_USE_R7
+    )
+endif()

--- a/tools/cmake/profiles/debug.cmake
+++ b/tools/cmake/profiles/debug.cmake
@@ -78,7 +78,6 @@ function(mbed_set_profile_options target mbed_toolchain)
         target_compile_definitions(${target}
             INTERFACE
                 __ASSERT_MSG
-                MULADDC_CANNOT_USE_R7
         )
     endif()
 

